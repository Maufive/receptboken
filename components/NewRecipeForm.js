import React, { Component } from "react";
import { Header, RecipeForm } from "./styles/Steg1Styles";
import Steg1 from "./Steg1";
import Steg2 from "./Steg2";
import Steg3 from "./Steg3";

class NewRecipeForm extends Component {
	state = {
		title: "",
		description: [],
		tags: [],
		timeRequired: 0,
		photo: "",
		ingredients: [],
		servings: 2,
		step: 1
	};

	saveToState = e => {
		this.setState({ [e.target.name]: e.target.value });
	};

	previousStep = () => {
		this.setState({ step: this.state.step - 1 });
	};

	saveIngredients = (arr, servings) => {
		this.setState({
			ingredients: arr,
			servings,
			step: this.state.step + 1
		});
	};

	saveDescription = arr => {
		this.setState({
			description: arr,
			step: this.state.step + 1
		});
	};

	saveDetails = (photo, time, tags) => {
		this.setState({
			photo,
			timeRequired: time,
			tags,
			step: this.state.step + 1
		});
	};

	render() {
		const step = this.state.step;
		let ActiveComponent;
		// För varje steg i processen att skapa ett nytt recept så renderar jag en specifik component.
		// Syftet är att dela upp en lång process att ladda upp ett recept i mindre bitar för att göra det
		// lättare för användare.
		if (step === 1) {
			ActiveComponent = (
				<Steg1
					saveToState={this.saveToState}
					saveIngredients={this.saveIngredients}
					ingredients={this.state.ingredients}
					servings={this.state.servings}
				/>
			);
		} else if (step === 2) {
			ActiveComponent = (
				<Steg2
					saveDescription={this.saveDescription}
					steg={step}
					previousStep={this.previousStep}
					description={this.state.description}
				/>
			);
		} else if (step === 3) {
			ActiveComponent = (
				<Steg3
					steg={step}
					previousStep={this.previousStep}
					saveDetails={this.saveDetails}
				/>
			);
		} else {
			ActiveComponent = <h2>Finished!</h2>;
			// To-do: Skriv en funktion för en POST-method att passa ner till steg 3. Spara Allt som finns i state här
			// till databasen.
			// Gör en LOADING komponent som snurrar medans man väntar på att databasen ska spara receptet
			// Gör en Page för ett recept där jag kan displaya informationen
			// Redirecta användaren när man lyckats spara ett recept till databasen till det receptet

		}

		return (
			<div>
				<Header>
					<input
						type="text"
						value={this.state.title}
						onChange={this.saveToState}
						name="title"
						required
						placeholder="Nytt recept..."
					/>
					<i onClick={this.editTitle} className="icofont-ui-edit" />
				</Header>
				<RecipeForm>{ActiveComponent}</RecipeForm>
			</div>
		);
	}
}

export default NewRecipeForm;
