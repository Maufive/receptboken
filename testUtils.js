export const fakeRecept = {
	_id: "1234",
	author: "5678436374re67fwuf76",
	created: "Idag",
	description: ["Steg1", "Steg2", "Steg3"],
	id: "1337",
	ingredients: [
		{
			input: "Lök",
			numberOfUnits: "2",
			units: "dl"
		}
	],
	photo: "asdf123.jpg",
	reviews: [{ score: 1 }],
	servings: "2",
	tags: ["Mat", "Gott"],
	timeRequired: "20 min",
	title: "Mitt jättegoda recept"
};

export const fakeUser = {
	_id: "1234",
	username: "fakeuser@gmail.com",
	password: "1234",
	hearts: [],
	fname: "Niklas",
	lname: "Albinsson",
	description: "Hej jag heter Niklas",
	photo: "http://fakephoto.jpg"
};

export const fakeShoppinglist = [
	{
		_id: "34567876543452176",
		title: "My Shoppinglist",
		created: "12/3 2019",
		author: "61386712376576d7q6ewd65",
		list: [
			{
				input: "Lök",
				numberOfUnits: "5",
				units: "dl"
			},
			{
				input: "Tomater",
				numberOfUnits: "10",
				units: "st"
			}
		]
	}
];
