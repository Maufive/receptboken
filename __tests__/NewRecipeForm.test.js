import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import NewRecipeForm from "../components/NewRecipeForm";
import { fakeUser } from "../testUtils";

describe("<NewRecipeForm />", () => {
	let wrapper;
	beforeEach(() => (wrapper = shallow(<NewRecipeForm user={fakeUser} />)));

	it("renders to the page and matches snapshot", () => {
		expect(wrapper.find("Steg1Test").length).toEqual(1);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	describe("Renders different components depending on Step-variable", () => {
		it("renders <Steg1 /> when Step is 1", () => {
			expect(wrapper.state().step).toEqual(1);
			expect(wrapper.find("#Steg1").length).toEqual(1);
		});

		it("renders <Steg2 /> when Step is set to 2", () => {
			wrapper.setState({ step: 2 });
			expect(wrapper.state().step).toEqual(2);
			expect(wrapper.find("#Steg2").length).toEqual(1);
		});

		it("renders <Steg3 /> when Step is set to 3", () => {
			wrapper.setState({ step: 3 });
			expect(wrapper.state().step).toEqual(3);
			expect(wrapper.find("#Steg3").length).toEqual(1);
		});
	});
});
