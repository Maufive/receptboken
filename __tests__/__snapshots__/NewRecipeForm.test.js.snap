// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewRecipeForm /> renders to the page and matches snapshot 1`] = `
<div>
  <div>
    <Steg1Test
      id="Steg1"
      ingredients={Array []}
      saveIngredients={[Function]}
      saveToState={[Function]}
      servings={2}
      title=""
    />
  </div>
</div>
`;
